src/invoke-lambda/spawn.js

```diff
@@ -54,44 +54,45 @@ module.exports = function spawnChild (params, callback) {
   // Ensure we don't have dangling processes due to open connections, etc.
   function maybeShutdown (event) {
     update.debug.status(`${functionPath} (pid ${pid}) shutting down (via ${event} event)`)
-    if (isRunning(pid) && !murderInProgress && !closed) {
-      update.debug.status(`${functionPath} (pid ${pid}) is still running, terminating now...`)
-      murderInProgress = true
-      let code = 0
-      if (error) {
-        update.error(`${functionPath} (pid ${pid}) caught hanging execution with an error, attempting to exit 1`)
-        code = 1
-      }
-      if (!isInLambda) {
-        kill(pid, SIG, () => {
-          murderInProgress = false
-          update.debug.status(`${functionPath} (pid ${pid}) successfully terminated`)
-          if (!closed) done(code)
-        })
-      }
-      else {
-        // tree-kill relies on *nix `ps`, which Lambda doesn't have – but it does have /proc
-        // Node process.kill() + Lambda Linux /proc/<pid>/task/<tid> is mysterious, so this may not be the best or proper approach
-        try {
-          let tasks = readdirSync(`/proc/${pid}/task`)
-          tasks.forEach(tid => {
-            try { process.kill(tid) }
-            catch (err) {
-              // Task may have ended naturally or been killed by killing child.pid, I guess we don't really know
-              update.debug.status(`${functionPath} (pid ${pid}) did not kill task (tid ${tid})`)
-            }
-          })
-          update.debug.status(`${functionPath} (pid ${pid}) (possibly maybe) successfully terminated inside Lambda`)
+    if (!isRunning(pid) || murderInProgress || closed) {
```

Comment:
Fully copping to the fact that I always struggle with these sorts of logical inversions, I'm not certain this is actually clearer for humans to read than what we have today?

I wouldn't make the claim that this code is incredibly clear (it's doing some funky stuff with some funky system stuff so it's just... funky), but the necessary mental gymnastics of the logical conditions for the early return are, at least for me, harder to follow than the alternative. That alternative being: only proceeding to the following lines if
- The process is still running, and
- The code below hasn't yet run (`murderInProgress === true`), and
- `done()` hasn't yet run (thus making `closed === false`)

Reply:
You know, looking at this again, and re-reading my comment / those predicates, maybe this just needs a code comment to prompt a realignment of the natural flow my brain seems to want to take for these kinds of predicate inversions.